upi/openstack/network.yaml

```diff
@@ -155,3 +155,61 @@
     ansible.builtin.command:
       cmd: "openstack floating ip set --port {{ os_port_ingress }} {{ os_ingress_fip }}"
     when: os_ingress_fip is defined and os_ingress_fip|length>0
+
+  - name: 'Create the primary cluster IPv6 subnet'
+    openstack.cloud.subnet:
+      name: "{{ os_subnet6 }}"
+      network_name: "{{ os_network }}"
+      cidr: "{{ os_subnet6_range }}"
+      ip_version: 6
+      ipv6_address_mode: "{{ os_subnet6_address_mode }}"
+      ipv6_ra_mode: "{{ os_subnet6_router_advertisements_mode }}"
+    when:
+    - os_subnet6_range is defined
+    - os_subnet6_range|ansible.utils.ipv6
+
+  - name: 'Add IPv6 subnet to the external router'
+    openstack.cloud.router:
+      name: "{{ os_router }}"
+      interfaces:
+      - "{{ os_subnet }}"
```

Comment:
Actually, when specifying interfaces, you need to provide previously added subnet, otherwise it will be assumed, that only `os_subnet6` should be set used and all the others removed. At my test env I got this, when no `os_subnet` is added to the list:
```
fatal: [localhost]: FAILED! => {"changed": false, "extra_data": {"data": null, "details": "Router interface for subnet cc4f26e0-c723-4e85-8b05-d9ea86394ef2 on router 8511e0ac-7c6a-4ee6-b731-42c0780eb32a cannot be deleted, as it is required by one or more floating IPs.", "response": "{\"NeutronError\": {\"type\": \"RouterInterfaceInUseByFloatingIP\", \"message\": \"Router interface for subnet cc4f26e0-c723-4e85-8b05-d9ea86394ef2 on router 8511e0ac-7c6a-4ee6-b731-42c0780eb32a cannot be deleted, as it is required by one or more floating IPs.\", \"detail\": \"\"}}"}, "msg": "ConflictException: 409: Client Error for url: https://10.1.8.125:13696/v2.0/routers/8511e0ac-7c6a-4ee6-b731-42c0780eb32a/remove_router_interface, Router interface for subnet cc4f26e0-c723-4e85-8b05-d9ea86394ef2 on router 8511e0ac-7c6a-4ee6-b731-42c0780eb32a cannot be deleted, as it is required by one or more floating IPs."}
```
which indicates that `openstack.cloud.router` is trying to remove the subnet v4 from router. Subnet `cc4f26e0-c723-4e85-8b05-d9ea86394ef2` is the primary subnet created at the beginning of this playbook.